keras_cv/metrics/object_detection/box_count_delta.py

```diff
@@ -0,0 +1,127 @@
+      name: (Optional) string name of the metric instance.
+      dtype: (Optional) data type of the metric result.
+
+    Usage:
+
+    ```python
+    box_count_delta = keras_cv.metrics.BoxCountDelta()
+    y_true = {
+        "classes": [[0, -1, -1]],
+        "boxes": [[[0, 0, 1, 1], [0, 0, 1, 1], [0, 0, 1, 1]]],
+    }
+    y_pred = {
+        "classes": [[0, 1, 1, -1]],
+        "boxes": [[[0, 0, 1, 1], [0, 0, 1, 1], [0, 0, 1, 1], [0, 0, 1, 1]]],
+    }
+    box_count_delta.update_state(y_true, y_pred)
+    print(box_count_delta.result())
+    # > 2.0
+    """
+
+    def __init__(
+        self,
+        mode="relative",
+        dtype=None,
+        name="box_count_delta",
+        **kwargs,
+    ):
+        super().__init__(
+            **kwargs,
+            dtype=dtype,
+            name=name,
+            **kwargs,
+        )
+        self.mode = mode
+        self.delta_sum = self.add_weight(
+            name="delta_sum",
+            shape=(),
+            dtype=self.compute_dtype,
+            initializer=initializers.Zeros(),
+        )
+        self.samples = self.add_weight(
+            name="samples",
+            shape=(),
+            dtype=self.compute_dtype,
+            initializer=initializers.Zeros(),
+        )
+
+    def reset_state(self):
+        self.delta_sum.assign(tf.zeros_like(self.delta_sum))
+        self.samples.assign(tf.zeros_like(self.samples))
+
+    def update_state(self, y_true, y_pred, sample_weight=None):
+        """
+        Args:
+            y_true: a bounding box Tensor in corners format.
+            y_pred: a bounding box Tensor in corners format.
+            sample_weight: Currently unsupported.
+        """
+        del sample_weight
+        y_true = bounding_box.ensure_tensor(y_true)
+        y_pred = bounding_box.ensure_tensor(y_pred)
+        y_pred = bounding_box.to_dense(y_pred)
+        y_true = bounding_box.to_dense(y_true)
+
+        ground_truth_boxes = tf.cast(y_true["classes"] != -1, tf.int32)
+        predicted_boxes = tf.cast(y_pred["classes"] != -1, tf.int32)
+
+        ground_truth_boxes = tf.math.reduce_sum(ground_truth_boxes, axis=-1)
+        predicted_boxes = tf.math.reduce_sum(predicted_boxes, axis=-1)
+
+        if self.mode == "relative":
+            delta_sum = predicted_boxes - ground_truth_boxes
+        elif self.mode == "absolute":
+            delta_sum = tf.math.abs(predicted_boxes - ground_truth_boxes)
+        else:
+            raise ValueError(
+                "`keras_cv.metrics.BoxCountDelta()` expects "
+                "either 'relative' or 'absolute' for `mode`. "
+                f"Received `mode={self.mode}`."
+            )
+
+        self.delta_sum.assign_add(
+            tf.cast(tf.math.reduce_sum(delta_sum), self.compute_dtype)
+        )
+        self.samples.assign_add(
+            tf.cast(tf.shape(y_pred["classes"])[0], self.compute_dtype)
+        )
+
+    def result(self):
+        return self.delta_sum / self.samples
```

Comment:
From a usefulness perspective, should we include stddev and mean as a dict here?

It seems like we could end up with a `mean` near `0` despite having a lot of extra and missing boxes, just in different images.

Your call, just an idea.

Reply:
I really like the idea of outputting the standard deviation.  I'll see if I can introduce this.